core/Columns/Dimension.php

```diff
@@ -747,7 +747,7 @@ protected function generateIdFromClass($className)
      */
     public static function getAllDimensions()
     {
-        $cacheId = CacheId::pluginAware('AllDimensions');
+        $cacheId = CacheId::siteAware(CacheId::pluginAware('AllDimensions'));
```

Comment:
I don't know if it is in another PR, but can be also make widgetsProvider siteAware and maybe reportsProvider? There is also `DimensionsProvider` class which needs to be siteAware (`getMapOfNameToDimension`). This is a new class that will at some point replace the old `Dimension::getAll()` as well.